File: samples/compose/app/src/main/kotlin/dev/getmilano/sample/designsystem/PrimaryButton.kt
package dev.getmilano.sample.designsystem

import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable

data class ButtonModel(
    val label: String,
    val isEnabled: Boolean = true,
    val onTap: () -> Unit = {},
)

@Composable
fun PrimaryButton(model: ButtonModel) {
    Button(onClick = model.onTap, enabled = model.isEnabled) {
        Text(model.label)
    }
}

File: samples/swiftui/Sources/DesignSystem/PrimaryButton.swift
import SwiftUI

struct ButtonModel {
    var label: String
    var isEnabled = true
    var onTap: () -> Void = {}
}

struct PrimaryButton: View {
    let model: ButtonModel

    var body: some View {
        Button(model.label, action: model.onTap)
            .buttonStyle(.borderedProminent)
            .disabled(!model.isEnabled)
    }
}
